### DISCUSSION TITLE
Is it possible for specific windows to remember their last position?

### DISCUSSION THREAD

**Discussion by ludgie** [labwc/labwc#1319] [Q&A] [answered]
Let's put the case where I always like my calculator app to spawn/appear on x/y coordinates or a defined region (rc.xml), is it possible or currently all windows can only spawn at center and not remember their last position? (or specify which ones can remember and which ones not)

**Comment by Consolatis** [comment, marked as answer]
X11 applications can technically remember their last position.
You can enforce a specific position for a window by using a [window rule](https://labwc.github.io/labwc-config.5.html#window_rules), something like this:
```xml
<windowRules>
	<windowRule identifier="XCalc">
		<!-- using specific coordinates -->
		<action name="MoveTo" x="20" y="20" />

		<!-- or alternatively using a defined region -->
		<!-- <action name="SnapToRegion" region="some_region" /> -->
	</windowRule>
</windowRules>
```

**Reply by ludgie** [reply]
Ok, tysm
